doc/developers/contributing.rst

```diff
@@ -619,16 +619,29 @@ Finally, follow the formatting rules below to make it consistently good:
         of the mentioned shapes. The default value is
         `np.ones(shape=(n_samples,))`.
 
+    list_param : list of int of shape (n_classes,)
```

Comment:
for builtin types it would have been nice to the typing system (i.e. `List[int]`. Basically we are re-inventing typing semantics here.

But I guess it's will be a bit more work to adopt and people not familiar with it might find it confusing.

Reply:
> sample_weight : ndarray[shape=(n_features,)], default=None

The problem is that as far as I know it's still no standard way of representing ndarray shape with types https://github.com/numpy/numpy/issues/7370#issuecomment-502884910

Also readability can indeed be challenging..